src/session/basic_group.rs

```diff
@@ -109,4 +127,73 @@ impl BasicGroup {
         self.imp().member_count.set(member_count);
         self.notify("member-count");
     }
+
+    pub(crate) fn status(&self) -> BoxedChatMemberStatus {
+        self.imp().status.borrow().as_ref().unwrap().to_owned()
+    }
+
+    pub(crate) fn set_status(&self, status: BoxedChatMemberStatus) {
+        if self.imp().status.borrow().as_ref() == Some(&status) {
+            return;
+        }
+        self.imp().status.replace(Some(status));
+        self.notify("status");
+    }
+
+    pub(crate) fn restriction_expression(
```

Comment:
Is this supposed to be a exact copy of Supergroup::restriction_expression`? If so, I'm not a fan of this. Also there is slight difference in the other method Supergroup:181

Reply:
> I'm not a fan of this
Sure, I can implement it in chat instead of both supergroup and basic_group

Reply:
so I did